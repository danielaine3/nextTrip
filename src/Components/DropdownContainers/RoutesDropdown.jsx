import React from 'react'
import { useHistory } from 'react-router-dom'
import { Dropdown } from '../Dropdown'

export const RoutesDropdown= ({routes, routeId, setRouteId}) => {
  const history = useHistory()
  if(!routes) return null
  
  return (
    <Dropdown 
      id='route-select'
      onSelect={(id) => {
        setRouteId(id)
        history.push(`/${id}`)
      }}
      options={routes}
      selected={routeId}
      title='Select Route'
    />
  )
}
