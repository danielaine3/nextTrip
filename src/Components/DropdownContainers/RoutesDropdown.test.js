import React from 'react'
import { render } from '@testing-library/react'
import { RoutesDropdown } from './RoutesDropdown'

const mockUseHistory = jest.fn()
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
    useParams: () => ({
      routeId: 'routeId',
      directionId: 'directionId',
      stopId: 'stopId'
    }),
    useHisory: () => ({ push: mockUseHistory })
  }
))

describe('<RoutesDropdown />', () => {
  it('should render null', async () => {
    const { asFragment } = await render(<RoutesDropdown />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render', async () => {
    const routes = [
      {route_id: "000", route_label: "METRO Blue Line"},
      {route_id: "111", route_label: "METRO Blue Line"}
    ]

    const { asFragment } = await render(
      <RoutesDropdown
        routes={routes} 
        routeId={'000'} 
        setRouteId={jest.fn()}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})