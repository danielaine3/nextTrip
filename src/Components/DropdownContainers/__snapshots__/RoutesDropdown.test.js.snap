// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoutesDropdown /> should render 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root makeStyles-formControl-1"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink="true"
      id="route-select-label"
    >
      Select Route
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="route-select-label route-select"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        id="route-select"
        role="button"
        tabindex="0"
      >
        METRO Blue Line
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value="000"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-4 PrivateNotchedOutline-legendNotched-5"
        >
          <span>
            Select Route
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<RoutesDropdown /> should render null 1`] = `<DocumentFragment />`;
